controllers/movies.ts

```diff
@@ -0,0 +1,302 @@
+import { Movie } from "../model/Movie";
+import { shortenURLs } from "../shorten-url";
+import { MovieParams, MovieQuerableParams } from "../util/types";
+
+class Movies {
+  async getAllMovies(params: MovieQuerableParams) {
+    const { actor, imdbSort, yearSort } = params;
+    try {
+      let query = Movie.createQueryBuilder("movie");
+      if (actor) {
+        query = query.where("movie.Actors LIKE :actor", {
+          actor: `%${actor}%`,
+        });
+      }
+      if (imdbSort) {
+        const imdbSortDirection =
+          imdbSort.toUpperCase() === "ASC" ? "ASC" : "DESC";
+        query = query.orderBy("movie.imdbRating", imdbSortDirection);
+      }
+
+      if (yearSort) {
+        const yearSortDirection =
+          yearSort.toUpperCase() === "ASC" ? "ASC" : "DESC";
+        query = query.orderBy("movie.Year", yearSortDirection);
+      }
+      let movies;
+
+      if (imdbSort) {
+        movies = await query
+          .select(["movie.Title", "movie.imdbRating"])
+          .getMany();
+      } else {
+        movies = await query.getMany();
+      }
+      return movies;
+    } catch (err: any) {
+      throw new Error("Movies not found");
+    }
+  }
+
+  async getMovieByImdID(imdb: any) {
+    try {
+      const movie = await Movie.findOne({
+        where: {
+          imdbID: imdb,
+        },
+      });
+      return movie;
+    } catch (err: any) {
+      throw new Error("Movie not found");
+    }
+  }
+
+  async deleteMovie(id: any) {
+    try {
+      const movie = (await Movie.findOne({
+        where: {
+          Id: id,
+        },
+      })) as Movie;
+      await Movie.remove(movie);
+      return movie;
+    } catch (err: any) {
+      throw new Error("Movie not found");
+    }
+  }
+
+  async addMovie(params: MovieParams) {
+    const {
+      Title,
+      Year,
+      Rated,
+      Released,
+      Runtime,
+      Genre,
+      Director,
+      Writer,
+      Actors,
+      Plot,
+      Language,
+      Country,
+      Awards,
+      Poster,
+      Metascore,
+      imdbRating,
+      imdbVotes,
+      imdbID,
+      Type,
+      Response,
+      Images,
+      ComingSoon,
+    } = params;
+
+    try {
+      const movie = await Movie.create({
+        Title: Title,
```

Comment:
You can just use `Title,` here since it is the same name and we are not doing any modifications. Same for the other parameters.

Reply:
I modified the method, it doesn't create new movie the same way anymore